Data/HashMap/Base.hs

```diff
@@ -1822,6 +1800,58 @@ updateOrConcatWithKey f ary1 ary2 = A.run $ do
     return mary
 {-# INLINABLE updateOrConcatWithKey #-}
 
+-- Helper functions for very rare cases.
+
+-- Remove (any) one element from an array of hashmaps
+unConsA :: A.Array (HashMap k v) -> UnCons k v
```

Comment:
Let's start over from scratch here. This all seems much too complicated! I'm thinking something like this:

```haskell
data UnconsHM k v
  = UnconsEmptyHM
  | UnconsedHM !(Leaf k v) !(HashMap k v)

data UnconsA k v
  = NowEmptyA !(Leaf k v)
  | UnconsedFromFirstA !(Leaf k v) !(A.Array (HashMap k v))
  | RemovedFirstA !(Leaf k v) !(A.Array (HashMap k v))

unconsA :: A.Array (HashMap k v) -> UnconsA k v
unconsA ary =
  case unconsHM  (A.index ary 0) of
    UnconsEmptyHM -> error "Unexpectedly empty internal map"
    UnconsedHM lf Empty
      | A.length ary == 1 -> NowEmptyA lf
      | otherwise -> -- Shorten the array and produce RemovedFirstA
    UnconsedHM lf hm' -> -- Replace the 0th element of the array with hm' and produce UnconsedFromFirstA
```

Reply:
Er... let me adjust that! We probably want to delete from the *last* one rather than the first one, because I imagine that makes the mask update easier.